src/systems/a320_systems/src/pneumatic.rs

```diff
@@ -1,27 +1,3055 @@
+    pack_pb_is_auto: bool,
+    pid: PidController,
+}
+impl PackFlowValveController {
+    fn new(context: &mut InitContext, engine_number: usize) -> Self {
+        Self {
+            pack_toggle_pb_id: context
+                .get_identifier(format!("AIRCOND_PACK{}_TOGGLE", engine_number)),
+            pack_pb_is_auto: true,
+            pid: PidController::new(0., 0.05, 0., 0., 1., 0.75),
+        }
+    }
+
+    fn update(&mut self, context: &UpdateContext, pack_flow_valve_flow_rate: VolumeRate) {
+        self.pid.next_control_output(
+            pack_flow_valve_flow_rate.get::<cubic_meter_per_second>(),
+            Some(context.delta()),
+        );
+    }
+}
+impl ControllerSignal<PackFlowValveSignal> for PackFlowValveController {
+    fn signal(&self) -> Option<PackFlowValveSignal> {
+        Some(match self.pack_pb_is_auto {
+            true => PackFlowValveSignal::new(Ratio::new::<ratio>(self.pid.output())),
+            false => PackFlowValveSignal::new_closed(),
+        })
+    }
+}
+impl SimulationElement for PackFlowValveController {
+    fn read(&mut self, reader: &mut SimulatorReader) {
+        self.pack_pb_is_auto = reader.read(&self.pack_toggle_pb_id);
+    }
+}
+
+/// This is a unique valve (and specific to the A320 probably) because it is controlled by two motors. One for manual control and one for automatic control
+pub struct CrossBleedValve {
+    open_amount: Ratio,
+    connector: PneumaticContainerConnector,
+    spring_characteristic: f64,
+    is_powered_for_manual_control: bool,
+    is_powered_for_automatic_control: bool,
+}
+impl CrossBleedValve {
+    pub fn new(spring_characteristic: f64) -> Self {
+        Self {
+            open_amount: Ratio::new::<ratio>(0.),
+            connector: PneumaticContainerConnector::new(),
+            spring_characteristic,
+            is_powered_for_manual_control: false,
+            is_powered_for_automatic_control: false,
+        }
+    }
+
+    pub fn update_move_fluid(
+        &mut self,
+        context: &UpdateContext,
+        from: &mut impl PneumaticContainer,
+        to: &mut impl PneumaticContainer,
+    ) {
+        if !self.is_powered_for_manual_control && !self.is_powered_for_automatic_control {
+            self.set_open_amount_from_pressure_difference(from.pressure() - to.pressure())
+        }
+
+        self.connector
+            .with_transfer_speed_factor(self.open_amount)
+            .update_move_fluid(context, from, to);
+    }
+
+    fn set_open_amount_from_pressure_difference(&mut self, pressure_difference: Pressure) {
+        self.open_amount = Ratio::new::<ratio>(
+            2. / PI
+                * (pressure_difference.get::<psi>() * self.spring_characteristic)
+                    .atan()
+                    .max(0.),
+        );
+    }
+
+    fn update_open_amount(&mut self, controller: &impl ControllerSignal<CrossBleedValveSignal>) {
+        if let Some(signal) = controller.signal() {
```

Comment:
I am not sure what happens in the real aircraft when the valve does not receive a signal from either BMC. I assume it will go into a pneumatic backup mode, where it is only controlled by physical forces due to the pressure differences between either side of the valve.

On a high level, when both BMCs fail in my code, this is also what will happen. But it's not because both send a `None` signal, it's because the only way for both BMCs to fail right now is the failure of the DC 2 and DC ESS SHED bus. In this case, the valve will also be unpowered and it will revert to pneumatic backup mode as soon as it realizes it's not powered, i.e when the electricity visitor comes for a visit.

I haven't given this too much thought to be honest because they are mostly all linked to failure scenarios. This is also a limitation of the PR (I will add it to the description).

I think it's a good concept that a valve can be connected to multiple controllers, where some controllers simply give a `None` signal which are ignored in favor of the other signals containing information. So I am leaning towards leaning this in, but I will have to think more concretely about the use cases.